Main.py

```diff
@@ -711,32 +851,103 @@ def __init__(self, controller, selection):
+        self.average_msgs_label.pack(side='top', pady=5)
+
+        self.avg_listbox = tk.Listbox(self, width=30, height=4)
+        self.avg_listbox.pack(side='top', pady=5)
+        self.avg_listbox.insert('end', f'{self.module.TITLE_PER_DAY} - {all_msgs / (sec_since_start / 86400):.2f}')
+        self.avg_listbox.insert('end', f'{self.module.TITLE_PER_WEEK} - {all_msgs / (sec_since_start / (7 * 86400)):.2f}')
+        self.avg_listbox.insert('end', f'{self.module.TITLE_PER_MONTH} - {all_msgs / (sec_since_start / (30 * 86400)):.2f}')
+        self.avg_listbox.insert('end', f'{self.module.TITLE_PER_YEAR} - {all_msgs / (sec_since_start / (365 * 86400)):.2f}')
+
+        # Add entry fields for message length filters
+        tk.Label(self, text='Min Length:').pack(side='top', pady=5)
+        self.min_length_entry = ttk.Entry(self)
+        self.min_length_entry.pack(side='top', pady=5)
+
+        tk.Label(self, text='Max Length:').pack(side='top', pady=5)
+        self.max_length_entry = ttk.Entry(self)
+        self.max_length_entry.pack(side='top', pady=5)
+
+        # Add a button to apply the filters
+        ttk.Button(self, text='Apply Filters', command=self.apply_filters).pack(side='top', pady=5)
+
+    def apply_filters(self):
+        # Fetch values from entry fields, applying defaults if necessary
+        try:
+            min_length = int(self.min_length_entry.get()) if self.min_length_entry.get().strip() else 0
+        except ValueError:
+            min_length = 0  # Use default if input is not a valid integer
 
-        listbox = tk.Listbox(self, width=30, height=4)
-        listbox.pack(side='top', pady=5)
-        listbox.insert('end', f'{self.module.TITLE_PER_DAY} - {all_msgs / (sec_since_start / 86400):.2f}')
-        listbox.insert('end', f'{self.module.TITLE_PER_WEEK} - {all_msgs / (sec_since_start / (7 * 86400)):.2f}')
-        listbox.insert('end', f'{self.module.TITLE_PER_MONTH} - {all_msgs / (sec_since_start / (30 * 86400)):.2f}')
-        listbox.insert('end', f'{self.module.TITLE_PER_YEAR} - {all_msgs / (sec_since_start / (365 * 86400)):.2f}')
+        try:
+            max_length = int(self.max_length_entry.get()) if self.max_length_entry.get().strip() else 10000000
+        except ValueError:
+            max_length = 10000000  # Use default if input is not a valid integer
+
+        # Update values in the MasterWindow
+        self.controller.min_message_length = min_length
+        self.controller.max_message_length = max_length
+
+        # Optionally refresh or re-fetch the data based on these new settings
+        self.refresh_data_based_on_length()
+
+    def refresh_data_based_on_length(self):
+        if self.controller.current_conversation:
+            filtered_data = self.controller.get_filtered_data(self.controller.current_conversation)
+            self.update_ui(filtered_data)
+        else:
+            print("No current conversation set to apply filters.")
```

Comment:
I think it's better to raise an exception, fail gracefully, or display an error dialogue rather than printing, since an end user won't be able to see this output.